components/brave_ads/core/internal/common/net/http/http_status_code_util.cc

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/common/net/http/http_status_code_util.h"
+
+#include "base/containers/fixed_flat_set.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr auto kAllowedHttpStatusCodes = base::MakeFixedFlatSet<int>({
+    400,  // Bad Request.
+    401,  // Unauthorized.
+    403,  // Forbidden.
+    404,  // Not Found.
+    408,  // Request Timeout.
```

Comment:
This one might be useful too.
```suggestion
    407,  // Proxy auth required.
    408,  // Request Timeout.
```

Reply:
Added, thanks